### PR TITLE
Update to androidx

### PR DESCRIPTION
**Pull request by ZacSweers**
This was a bit of a journey to bring the project into 2020.

- API has not changed, only androidx and dependencies are updated
- Releases are now done via the [maven gradle publish plugin](https://github.com/vanniktech/gradle-maven-publish-plugin) (same used by a number of other popular libraries). It's mostly the same command, but now you just run `./gradlew :jobqueue:uploadArchives --no-parallel`. You'll need to tweak your local sonatype gradle properties to match the new, upper-case versions (see gradle.properties)
- Project is structurally modernized. Single gradle root project with the library and sample being subprojects.
- I had to hack a lot to get greenrobot libraries to work. For greendao in particular, I just manually pasted in the generated code and called it a day there.
- GMS is now a `compileOnly` dependency, which achieves the prior `provided` goal
- Jobqueue only ships a dependency on okio now. Its actual androidx usage was just limited to (maybe) annotations. It had a support v4 dependency, but that was just for `Pair` and only used in tests.

Tests all pass for me locally, but one of the GMS ones appears to be a bit flaky (will check which one).

### CODE REVIEW

**Comment by ZacSweers** [comment]
The same app _runs_, but only as far as it'll attempt to make a twitter API call. I don't have credentials, or the API it uses no longer works. ¯\_(ツ)_/¯.

**Comment by ZacSweers** [comment]
Was hoping github actions would run since circle wasn't seeming to, but I guess no status checks are running?

**Comment by ZacSweers** [comment]
Here's the run on actions on my repo. Test failure there appears to be flaky, as it passes locally for me https://github.com/ZacSweers/android-priority-jobqueue/runs/398537325

File: examples/twitter/TwitterClient/src/main/java/com/birbit/android/jobqueue/examples/twitter/dao/DaoMaster.java [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import android.database.sqlite.SQLiteDatabase.CursorFactory;
```
Comment (yigit) [inline examples/twitter/TwitterClient/src/main/java/com/birbit/android/jobqueue/examples/twitter/dao/DaoMaster.java]: guess i should eventually replace this w/ Room. :p

File: gradle.properties [hunk 2]
```diff
@@ -0,0 +1,23 @@
+android.enableJetifier=true
```
Comment (yigit) [inline gradle.properties]: we don't need this, do we ?

Comment (ZacSweers) [reply gradle.properties]: It's necessary for the sample project for now unfortunately, greenrobot's libraries still think it's 2018.

File: jobqueue/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -8,4 +8,5 @@
     com.google.android.gms, com.google.android.gms.measurement"/>
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
```
Comment (virusman) [inline jobqueue/src/main/AndroidManifest.xml:9]: Why is this required now?
